STYLE.md

```diff
@@ -0,0 +1,270 @@
+
+Inspired by [The Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide) and [The Puppet Style Guide](https://docs.puppetlabs.com/guides/style_g uide.html).
+
+## Syntax
+
+- Strings are in double-quotes.
+
+### Spacing
+
+Use 2 spaces when defining resources except when defining inline policies or other inline resources.
+
+```
+resource "aws_iam_role" "iam_role" {
+  name = "${var.resource_name}-role"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+```
+
+There should be two blank lines between resource definitions.
+
+```hcl
+// bad
+variable "vpc_id" {}
+
+variable "vpc_id" {
+  description = "The VPC this security group will go in"
+  type        = "string"
+}
+
+// goo
+variable "vpc_id" {
+  description = "The VPC this security group will go in"
+  type        = "string"
+}
+
+
+variable "public_subnet_ids" {
+  description = "List of public subnet IDs"
+  type        = "list"
+  default     = []
+}
+```
+
+
+
+### Resource Block Alignment
+
+Parameter definitions in a resource block should be aligned. The `terraform fmt` command can do this for you.
+
+```hcl
+// bad
+resource "aws_security_group" "main" {
+    name = "${var.name}"
+    description = "Security Group ${var.name}"
+    vpc_id = "${var.vpc_id}"
+    tags {
+        Name = "${var.name}"
+    }
+}
+
+// good
+resource "aws_security_group" "main" {
+  name        = "${var.name}"
+  description = "Security Group ${var.name}"
+  vpc_id      = "${var.vpc_id}"
+  tags {
+    Name = "${var.name}"
+  }
+}
+```
+### Variables
+
+Variables should be provided in a `variables.tf` file at the root of your module.
+
+A description and type should be provided for each declared variable. When it makes sense, please provide a reasonable default value.
+
+## Outputs
+
+Outputs should be provided in an `outputs.tf` file at the root of your module.
+
+Attributes exported by your modules should follow the same descriptive format as variable names.
+
+## Comments
+
+Only comment what is necessary.
+
+Single line comments: `#` or `//`.
+
+Multi-line comments: `/*` followed by `*/`.
+
+When commenting use two "//" and a space in front of the comment.
+
+```
+// CREATE ELK IAM ROLE 
+...
+```
+
+## Naming Conventions
+
+### File Names
+
+Create a separate resource file for each type of AWS resource. Similar resources should be defined in the same file and named accordingly.
```

Comment:
Don't agree with this at all. I find the code harder to follow when it's split up this way. In my opinion we should start with `main.tf` and only split things out when it gets hard to read, and at this point it's usually enough to move policy declarations to a standalone file (e.g. `policies.tf`).